src/main/scala/com/codacy/parsers/implementation/CoberturaParser.scala

```diff
@@ -46,23 +39,22 @@ object CoberturaParser extends CoverageParser {
       lineCoverage(cleanFilename, classes)
     })(collection.breakOut)
 
-    Right(CoverageReport(total, fileReports))
+    CoverageReport(total, fileReports)
   }
 
   private def lineCoverage(sourceFilename: String, classes: NodeSeq): CoverageFileReport = {
-    val classHit = (classes \\ "@line-rate").map { total =>
+    val classHit = (classes \\ LineRateAttribute).map { total =>
       val totalValue = TextUtils.asFloat(total.text)
-      (totalValue * 100).toInt
+      math.round(totalValue * 100)
     }
-    val fileHit = if (classHit.length != 0) { classHit.sum / classHit.length } else 0
+    val fileHit = if (classHit.nonEmpty) { classHit.sum / classHit.length } else 0
 
     val lineHitMap: Map[Int, Int] =
       (for {
         xClass <- classes
         line <- xClass \\ "line"
-      } yield (line \ "@number").text.toInt -> (line \ "@hits").text.toInt)(collection.breakOut)
+      } yield (line \ "@number").text.toInt -> (line \ "@hits").text.toInt).toMap
```

Comment:
Here you can use `line \@ "number"`
```suggestion
      } yield (line \@ "number").toInt -> (line \ "@hits").text.toInt).toMap
```

Reply:
Done ✅